File: 백준/Silver/2615. 오목/오목.cc
#include <iostream>
using namespace std;

int dy[] = {-1, 0, 1, 1};
int dx[] = {1, 1, 1, 0};
int board[19][19];
int win_y;
int win_x;

int successive(int y, int x) {
    int color = board[y][x];
    for (int i = 0; i < 4; i++) {
        int py = y - dy[i];
        int px = x - dx[i];
        if (py >= 0 && px >= 0 && py < 19 && px < 19 && board[py][px] == color) continue;
        int ny = y;
        int nx = x;
        int count = 1;
        for (int j = 0; j < 6; j++) {
            ny += dy[i];
            nx += dx[i];
            if (ny >= 0 && nx >= 0 && ny < 19 && nx < 19) {
                if (board[ny][nx] == color) {
                    count++;
                }
                else break;
            }
        }
        if (count == 5) return color;
        
    }
    return 0;
}

int check() {
    for (int i = 0; i < 19; i++) {
        for (int j = 0; j < 19; j++) {
            int res = successive(i, j);
            if (res) {
                win_y = i;
                win_x = j;
                return res;
            }
        }
    }
    return 0;
}

int main() {
    for (int i = 0; i < 19; i++) for (int j = 0; j < 19; j++) cin >> board[i][j];
    int winner = check();
    if (winner) printf("%d\n%d %d", winner, win_y+1, win_x+1);
    else cout << 0;
    return 0;
}

File: 백준/Silver/2615. 오목/오목.py
def is_game_over(r, c):
    global win_y
    global win_x
    color = board[r][c]
    dr = [0, 1, 1, -1]
    dc = [1, 1, 0, 1]
    for di in range(4):
        pr = r - dr[di]
        pc = c - dc[di]
        if 0 <= pr < 19 and 0 <= pc < 19 and board[pr][pc] == color:
            continue
        nr = r
        nc = c
        length = 1
        for _ in range(5):
            nr += dr[di]
            nc += dc[di]
            if 0 <= nr < 19 and 0 <= nc < 19:
                if board[nr][nc] == color:
                    length += 1
                else:
                    break
        if length == 5:
            win_y = r + 1
            win_x = c + 1
            return color
    return 0


def search():
    for y in range(19):
        for x in range(19):
            if board[y][x] > 0:
                winner = is_game_over(y, x)
                if winner:
                    return winner


board = []
for _ in range(19):
    row = list(map(int, input().split()))
    board.append(row)
win_y = 0
win_x = 0
ans = search()
if ans:
    print(ans)
    print(win_y, win_x)
else:
    print(0)
